core/include/scipp/core/multi_index.h

```diff
@@ -193,6 +232,35 @@ template <scipp::index N> class SCIPP_CORE_EXPORT MultiIndex {
     return m_coord != other.m_coord;
   }
 
+  constexpr bool in_same_chunk(const MultiIndex &other,
+                               const scipp::index first_dim) const noexcept {
+    if (m_ndim != other.m_ndim) {
+      return false;
+    }
```

Comment:
Is this check necessary? It would catch some cases of comparing iterators from different containers, but that is generally UB anyway?

Reply:
It's not necessary. So having some UB in these cases is fine?

Reply:
I think my point was that if this method ever gets called with such different iterators we would have UB in *other* places.